Fix ignoring return value warning in WPEDisplayDRM::findDevice()

psaavedra opened the pull request:
#### 05b8197bbdfcb28d644ef45d3d468432437ac6b3
<pre>
Fix ignoring return value warning in WPEDisplayDRM::findDevice()
<a href="https://bugs.webkit.org/show_bug.cgi?id=268544">https://bugs.webkit.org/show_bug.cgi?id=268544</a>

Reviewed by Carlos Garcia Campos.

The 273896@main introduces an wrong use of fd.release(). This
leads in a build warning error and a potential file descriptor leak.
Use &apos;fd = { };&apos; instead of this to reset as invalid.

* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(findDevice):

Canonical link: <a href="https://commits.webkit.org/273902@main">https://commits.webkit.org/273902@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/083cc9aefa8d794e42f734fba550e53ac207ee4b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37155 "Failed to checkout and rebase branch from PR 23662") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16057 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39454 "Built successfully") | [❌ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39728 "Failed to checkout and rebase branch from PR 23662") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/33164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18635 "Built successfully") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/51/builds/13198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/5/builds/39728 "Failed to checkout and rebase branch from PR 23662") | 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/37718 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/13553 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/32712 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/5/builds/39728 "Failed to checkout and rebase branch from PR 23662") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/11803 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/40985 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/33650 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/33682 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/40985 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/12131 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/9924 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/40985 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/13754 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/12488 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4808 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/13018 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
```diff
@@ -95,7 +95,7 @@ static UnixFileDescriptor findDevice(GError** error)
             break;
 
-        fd.release();
+        std::ignore = fd.release();
```
This is leaking the fd, just remove the release().

carlosgcampos commented on Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
```diff
@@ -95,7 +95,8 @@ static UnixFileDescriptor findDevice(GError** error)
 
-        fd.release();
+        fd = { };
+
```
Remove this empty line, please

psaavedra committed:
Fix ignoring return value warning in WPEDisplayDRM::findDevice()
https://bugs.webkit.org/show_bug.cgi?id=268544

Reviewed by Carlos Garcia Campos.

The 273896@main introduces an wrong use of fd.release(). This
leads in a build warning error and a potential file descriptor leak.
Use 'fd = { };' instead of this to reset as invalid.

* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(findDevice):

Canonical link: https://commits.webkit.org/273902@main
